server/src/main/java/org/opensearch/cluster/awarenesshealth/ClusterAwarenessHealth.java

```diff
@@ -0,0 +1,119 @@
+
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.routing.allocation.AwarenessReplicaBalance;
+import org.opensearch.cluster.routing.allocation.decider.AwarenessAllocationDecider;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Settings;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Objects;
+
+/**
+ * Cluster state Awareness health information
+ *
+ */
+public class ClusterAwarenessHealth implements Writeable {
+
+    private final ClusterAwarenessAttributesHealth clusterAwarenessAttributesHealth;
+
+    /**
+     * Creates cluster awareness health from cluster state.
+     *
+     * @param clusterState The current cluster state. Must not be null.
+     * @param clusterSettings the current cluster settings.
+     * @param awarenessAttributeName Name of awareness attribute for which we need to see the health
+     */
+    public ClusterAwarenessHealth(
+        ClusterState clusterState,
+        ClusterSettings clusterSettings,
+        String awarenessAttributeName,
+        int totalShards
+    ) {
+
+        // This property will govern if we need to show unassigned shard info or not
+        boolean displayUnassignedShardLevelInfo = canCalcUnassignedShards(clusterSettings, awarenessAttributeName);
+
+        clusterAwarenessAttributesHealth = new ClusterAwarenessAttributesHealth(
+            awarenessAttributeName,
+            displayUnassignedShardLevelInfo,
+            clusterState,
+            totalShards
+        );
+    }
+
+    private boolean canCalcUnassignedShards(ClusterSettings clusterSettings, String awarenessAttributeName) {
+
+        // Getting the replicaEnforcement settings as both are necessary for replica enforcement.
+        boolean allocationAwarenessBalance = clusterSettings.get(
+            AwarenessReplicaBalance.CLUSTER_ROUTING_ALLOCATION_AWARENESS_BALANCE_SETTING
+        );
+        Settings forcedAwarenessSettings = clusterSettings.get(
+            AwarenessAllocationDecider.CLUSTER_ROUTING_ALLOCATION_AWARENESS_FORCE_GROUP_SETTING
+        );
+
+        boolean forcedZoneSettingsExists = false;
+
+        if (!forcedAwarenessSettings.isEmpty()) {
+            // We will only mark true if particular awareness attribute exists
+            if (forcedAwarenessSettings.hasValue(awarenessAttributeName + ".values")) {
+                forcedZoneSettingsExists = true;
+            }
+        }
+
+        return allocationAwarenessBalance && forcedZoneSettingsExists;
+    }
+
+    public ClusterAwarenessHealth(final StreamInput in) throws IOException {
+        clusterAwarenessAttributesHealth = new ClusterAwarenessAttributesHealth(in);
+    }
+
+    public ClusterAwarenessHealth(String awarenessAttribute) {
```

Comment:
We have two constructors in each of `ClusterAwarenessAttributesHealth` and `ClusterAwarenessHealth` are they meant for test only?

Reply:
Yes that is correct using for unit test cases